### PR TITLE
reference: add PD scheduling best practice

### PR DESCRIPTION
**Pull request by dcalvin**
### What is changed, added or deleted? 

This PR added the PD Scheduling Best Practices to the Best Practice series. 

### Which version does your change affect? 

TiDB 3.0/Dev/2.1

Only added for Dev for now.

### CODE REVIEW

File: dev/glossary.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+---
+title: 术语表
+summary: 学习 TiDB 相关术语
+category: glossary
+---
+
+# 术语表
```
Comment (disksing) [inline dev/glossary.md]: 👍 好主意，不过另外开个 PR 搞这个合适点？这里面缺了很多重要的术语/概念

Comment (dcalvin) [reply dev/glossary.md]: 嗯，这个只是一个 trigger，占个地方，后续都往这个文件添加。

Comment (disksing) [reply dev/glossary.md]: 放到另外的PR吧。这篇文章不要术语表也能work的。

File: dev/glossary.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+### Leader/Follower/Learner
+
+它们分别对应 [Peer](#regionpeerraft-group) 的三种角色。其中 Leader 负责响应客户端的读写请求；Follower 被动地从 Leader 同步数据，当 Leader 失效时会进行选举产生新的 Leader；Learner 是一种特殊的角色，它只参与同步 raft log 而不参与投票，在目前的实现中只短暂存在于添加副本的中间步骤。
+
+## O
+
+### Operator
+
+Operator 是应用于一个 Region 的，服务于某个调度目的的一系列操作的集合。例如“将 Region 2 的 Leader 迁移至 Store 5”，“将 Region 2 的副本迁移到 Store 1, 4, 5” 等。
```
Comment (lilin90) [inline dev/glossary.md]:
```suggestion
Operator 是应用于一个 Region 的，服务于某个调度目的的一系列操作的集合。例如“将 Region 2 的 Leader 迁移至 Store 5”，“将 Region 2 的副本迁移到 Store 1, 4, 5”等。
```

File: dev/glossary.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+Operator 可以是由 Scheduler 通过计算生成的，也可以是由外部 API 创建的。
+
+### Operator Step
+
+Operator Step 是 Operator 执行过程的一个步骤，一个 Operator 常常会包含多个 Operator Step。
+
+目前 PD 可生成的 Step 包括：
+
+- TransferLeader：将 Region Leader 迁移至指定 Peer
```
Comment (lilin90) [inline dev/glossary.md]:
```suggestion
- `TransferLeader`：将 Region Leader 迁移至指定 Peer
```

File: dev/glossary.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+- AddPeer：在指定 Store 添加 Follower
```
Comment (lilin90) [inline dev/glossary.md]:
```suggestion
- `AddPeer`：在指定 Store 添加 Follower
```

File: dev/glossary.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+- RemovePeer：删除一个 Region Peer
```
Comment (lilin90) [inline dev/glossary.md]:
```suggestion
- `RemovePeer`：删除一个 Region Peer
```

File: dev/glossary.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+- AddLearner：在指定 Store 添加 Region Learner
```
Comment (lilin90) [inline dev/glossary.md]:
```suggestion
- `AddLearner`：在指定 Store 添加 Region Learner
```

File: dev/glossary.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+- PromoteLearner：将指定 Learner 提升为 Follower
```
Comment (lilin90) [inline dev/glossary.md]:
```suggestion
- `PromoteLearner`：将指定 Learner 提升为 Follower
```

File: dev/glossary.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+- SplitRegion：将指定 Region 一分为二
```
Comment (lilin90) [inline dev/glossary.md]:
```suggestion
- `SplitRegion`：将指定 Region 一分为二
```

File: dev/glossary.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+其机制是集群初始化时构建一个初始 Region 覆盖整个 key space，随后在运行过程中每当 Region 数据达到一定量之后就通过 Split 产生新的 Region。
+
+## S
+
+### Scheduler
+
+Scheduler（调度器）是 PD 中生成调度的组件。PD 中每个调度器是独立运行的，分别服务于不同的调度目的。常用的调度器及其调用目标有：
+
+- balance-leader-scheduler：保持不同节点的 Leader 均衡。
```
Comment (lilin90) [inline dev/glossary.md]:
```suggestion
- `balance-leader-scheduler`：保持不同节点的 Leader 均衡。
```

File: dev/glossary.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+- balance-region-scheduler：保持不同节点的 Peer 均衡。
```
Comment (lilin90) [inline dev/glossary.md]:
```suggestion
- `balance-region-scheduler`：保持不同节点的 Peer 均衡。
```

File: dev/glossary.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+- hot-region-scheduler：保持不同节点的读写热点 Region 均衡。
```
Comment (lilin90) [inline dev/glossary.md]:
```suggestion
- `hot-region-scheduler`：保持不同节点的读写热点 Region 均衡。
```

File: dev/glossary.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+- evict-leader-{store-id}：驱逐某个节点的所有 Leader。（常用于滚动升级）
```
Comment (lilin90) [inline dev/glossary.md]:
```suggestion
- `evict-leader-{store-id}`：驱逐某个节点的所有 Leader。（常用于滚动升级）
```

File: dev/glossary.md [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+
+### Store
+
+PD 中的 Store 指的是集群中的存储节点，也就是 tikv-server 实例。 Store 与 TiKV 实例是严格一一对应的，即使在同一主机甚至同一块磁盘部署多个 TiKV 实例，这些实例也对会对应不同的 Store。
```
Comment (lilin90) [inline dev/glossary.md]:
```suggestion
PD 中的 Store 指的是集群中的存储节点，也就是 tikv-server 实例。Store 与 TiKV 实例是严格一一对应的，即使在同一主机甚至同一块磁盘部署多个 TiKV 实例，这些实例也对会对应不同的 Store。
```

**Comment by lilin90** [comment]
@dcalvin Please address comments. Thanks!

File: dev/glossary.md (hunk 1 above)
Comment (dcalvin) [reply dev/glossary.md]: 又讨论了下，先保留啦，也不影响～

**Comment by dcalvin** [comment]
@anotherrachel FYI. You can start the translation.

**Comment by lilin90** [comment]
@dcalvin The new file wasn't added to TOC in v3.0 & v2.1. Please add it.

**Comment by lilin90** [comment]
@anotherrachel Please combine this PR with #2003 when you align docs. Thanks!

**Pull request merged**